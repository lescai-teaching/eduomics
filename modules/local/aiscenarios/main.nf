process AISCENARIOS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/pip_google-genai:842241e3cb24ce31':
        'community.wave.seqera.io/library/pip_google-genai:657c5e158292b26a' }"

    input:
    tuple val(meta), val(variant)
    tuple val(meta2), val(genes)

    output:
    tuple val(meta), path("*_scenario.txt"), emit: scenario
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def script_command = variant ? "generate_variant_scenarios.py --variant ${variant} --output ${prefix}_scenario.txt" : "generate_differentialexpression_scenarios.py --genes ${genes} --output ${prefix}_scenario.txt"

    """
    $script_command

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        aiscenarios: 1.0.0)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def script_command = variant ? "./generate_variant_scenarios --variant ${variant} --output ${prefix}_scenario.txt" : "generate_differentialexpression_scenario --genes ${genes} --output ${prefix}_scenario.txt"

    """
    touch ${prefix}_scenario.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        aiscenarios: 1.0.0)
    END_VERSIONS
    """
}
