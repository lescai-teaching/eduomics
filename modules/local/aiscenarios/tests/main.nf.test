
nextflow_process {

    name "Test Process AISCENARIOS"
    script "../main.nf"
    config "./env.config"
    process "AISCENARIOS"

    tag "modules"
    tag "modules_"
    tag "aiscenarios"

    test("scenario with variant") {

        when {
            process {
                """
                input[0] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    "chr22-18078561-G-A",
                ]

                input[1] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    false,
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.scenario.get(0).get(1).endsWith('_scenario.txt') },
                { assert process.out.versions }

            )
        }

    }

    test("scenario with genes") {

        when {
            process {
                """
                input[0] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    false,
                ]

                input[1] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    "TCN2,NAGA,CYB5R3,PNPLA5,ARSA",
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.scenario.get(0).get(1).endsWith('_scenario.txt') },
                { assert process.out.versions }

            )
        }

    }

    test("scenario - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    false,
                ]

                input[1] = [
                    [ id:'simu01', coverage:50 ], // meta map
                    "TCN2,NAGA,CYB5R3,PNPLA5,ARSA",
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.scenario.get(0).get(1).endsWith('_scenario.txt') },
                { assert process.out.versions }

            )
        }

    }

}
