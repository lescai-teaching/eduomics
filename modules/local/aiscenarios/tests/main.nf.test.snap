{
    "scenario with genes": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-01T08:34:18.288383158"
    },
    "scenario - stub": {
        "content": [
            [
                [
                    {
                        "id": "simu01",
                        "coverage": 50
                    },
                    "simu01_scenario.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-01T08:34:29.485921161"
    },
    "scenario with variant": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-01T08:33:57.014297644"
    }
}